symengine/number.h

```diff
@@ -15,7 +15,9 @@ class Evaluate;
 
 class Number : public Basic {
 public:
-    //! \return true if `0`
+    //! \return true if exactly equal to `0`
+    virtual bool is_exact_zero() const = 0;
+    //! \return true if equal to `0`
     virtual bool is_zero() const = 0;
```

Comment:
Can you describe in the documentation here what the difference between `is_exact_zero()` and `is_zero()` is and when one should use each?

Also, we need to add tests for these two functions. We only test this indirectly so far.